src/services/repo/useRepoSettings.tsx

```diff
@@ -0,0 +1,151 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+import { RepoNotFoundErrorSchema } from './schemas/RepoNotFoundError'
+import { RepoOwnerNotActivatedErrorSchema } from './schemas/RepoOwnerNotActivatedError'
+
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  private: z.boolean().nullable(),
+  activated: z.boolean().nullable(),
+  uploadToken: z.string().nullable(),
+  defaultBranch: z.string().nullable(),
+  profilingToken: z.string().nullable(),
+  staticAnalysisToken: z.string().nullable(),
+  graphToken: z.string().nullable(),
+  yaml: z.string().nullable(),
+  bot: z
+    .object({
+      username: z.string().nullable(),
+    })
+    .nullable(),
+})
+
+interface FetchRepoSettingsArgs {
+  provider: string
+  owner: string
+  repo: string
+  signal?: AbortSignal
+}
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+function fetchRepoSettingsDetails({
+  provider,
+  owner,
+  repo,
+  signal,
+}: FetchRepoSettingsArgs) {
+  const query = `
```

Comment:
Can we pull the query out of the hook? Prevent a couple re-declarations there and clean up the logic a bit